src/components/icon/icon-state.broadcast.ts

```diff
@@ -0,0 +1,89 @@
+import { internalIcons } from './internal-icons-lib.js';
+import { SvgIconParser } from './registry/parser.js';
+import type {
+  BroadcastIconsChangeMessage,
+  Collection,
+  IconMeta,
+  SvgIcon,
+} from './registry/types.js';
+import { ActionType } from './registry/types.js';
+
+export class IconsStateBroadcast {
+  private parser: SvgIconParser;
+  private iconBroadcastChannel: BroadcastChannel;
+  constructor(
+    collections: Collection<string, Map<string, SvgIcon>>,
+    refsCollection: Collection<string, Map<string, IconMeta>>
+  ) {
+    this.parser = new SvgIconParser();
+    this.iconBroadcastChannel = new BroadcastChannel('ignite-ui-icon-channel');
+    this.iconBroadcastChannel.onmessage = (event) => {
+      const message = event.data as BroadcastIconsChangeMessage;
+      if (message.actionType === ActionType.SyncState) {
```

Comment:
I suppose this work in progress. Don't we have to update the state of the icon registry when we receive `RegisterIcon`/`UpdateIconReference`. 
I guess we also have to update the state in the initial sync as well since this registry can be created after the angular icon service is already up and having a few custom icons.

Reply:
Do we actually need that? There's no public API exposed from angular element's side that would allow us to update the icons. So how would we change them from the angular icon service?

Reply:
Fair point. I was completely absent-minded that the angular elements stuff relies on the WC public API 💀

Then in that case can you update the tests to only check for the initial sync status on another peer. Maybe add a couple around the icon refs event